src/plugins/console/README.md

```diff
@@ -2,4 +2,55 @@
 
 ## About
 
-Console provides the user with tools for storing and executing requests against Elasticsearch.
\ No newline at end of file
+Console provides the user with tools for storing and executing requests against Elasticsearch.
+
+## Architecture
+- Ace editor wrapped with `<KIBANA_REPO>/src/plugins/console/public/types/core_editor.ts`. This wrapper allows for an easier change of the editor, for example from Ace to Monaco.
+- Autocomplete logic in `<KIBANA_REPO>/src/plugins/console/public/lib/autocomplete`. Autocomplete rules are computed by classes in `components` folder.
+
+## Autocomplete definitions
+Users benefit greatly from autocomplete suggestions since not all Elasticsearch APIs can be provided with a corresponding UI. The autocomplete definitions are all created in the form of javascript objects loaded from `json` or `js` files. 
+
+### Creating definitions
+The folder `<KIBANA_REPO>/src/plugins/console/server/lib/spec_definitions/json/generated` contains definitions generated automatically from Elasticsearch REST API specifications. See `<KIBANA_REPO>/packages/kbn-spec-to-console/README.md` for more information on the `spec-to-console` script. 
+
+Manually created override files in the folder `<KIBANA_REPO>/src/plugins/console/server/lib/spec_definitions/json/overrides` contain fixes for generated files and additions for body request parameters.   
+
+### Top level keys
+Following top level keys are used in the definitions objects:
+
+- `documentation`: string for the docs url link, `master` and `current` strings in the url are replaced with the `docLinkVersion`
+- `methods`: array of strings for allowed http methods
+- `patterns`: array of strings for url endpoints, accept parameters like `{indices}`, `{fields}` etc
+- `url_params`: an object with possible keys for url parameters and its values
+- `priority`: a number to decide when there are several possible autocomplete definitions
+- `data_autocomplete_rules`: an object with possible keys for body request and its values
+
+### Url parameters
+When defining url parameters or body request parameters, each key in the object is a possible url parameter or a possible field in the body request. Parameters like `{indices}`, `{fields}` etc are accepted both as a key and a value.
+
+- `__flag__`: is used for boolean params, suggestions contain `true` and `false`
+- array with a list of values: is used for a list of known accepted param values, suggestions contain all values in the list
+
+### Body request params:
+- `__one_of: [..., ...]`: is used for a list of known accepted param values, autocompletes as the first value of the array, suggestions contain all values in the list
+- `__one_of: [true, false]`: is used for boolean body request params
+- `__any_of: [..., ...]`: is used for a list of known accepted param values, autocompletes with an empty array, suggestions contain all values in the list
```

Comment:
This seems identical to `__one_of`. Is there any difference?

The end of the description states "is the same as `__any_of: [..., ...]`", but I'm not sure what this means since it seems to state "X is the same as X".